#! /bin/sh /usr/share/dpatch/dpatch-run
## 32_vscroll.dpatch by Junichi Uekawa <dancer@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: remove scrollbar patch, updated from 247602 by Eugeniy Meshcheryakov <eugen@univ.kiev.ua>

@DPATCH@
diff -urNad xpdf-3.01~/xpdf/XPDFCore.cc xpdf-3.01/xpdf/XPDFCore.cc
--- xpdf-3.01~/xpdf/XPDFCore.cc	2006-07-13 00:06:46.000000000 +0900
+++ xpdf-3.01/xpdf/XPDFCore.cc	2006-07-13 00:08:17.000000000 +0900
@@ -881,39 +881,40 @@
   XtSetArg(args[n], XmNvisualPolicy, XmVARIABLE); ++n;
   scrolledWin = XmCreateScrolledWindow(parentWidget, "scroll", args, n);
   XtManageChild(scrolledWin);
-  n = 0;
-  XtSetArg(args[n], XmNorientation, XmHORIZONTAL); ++n;
-  XtSetArg(args[n], XmNminimum, 0); ++n;
-  XtSetArg(args[n], XmNmaximum, 1); ++n;
-  XtSetArg(args[n], XmNsliderSize, 1); ++n;
-  XtSetArg(args[n], XmNvalue, 0); ++n;
-  XtSetArg(args[n], XmNincrement, 1); ++n;
-  XtSetArg(args[n], XmNpageIncrement, 1); ++n;
-  hScrollBar = XmCreateScrollBar(scrolledWin, "hScrollBar", args, n);
-  XtManageChild(hScrollBar);
-  XtAddCallback(hScrollBar, XmNvalueChangedCallback,
-		&hScrollChangeCbk, (XtPointer)this);
+  if (!fullScreen) {
+    n = 0;
+    XtSetArg(args[n], XmNorientation, XmHORIZONTAL); ++n;
+    XtSetArg(args[n], XmNminimum, 0); ++n;
+    XtSetArg(args[n], XmNmaximum, 1); ++n;
+    XtSetArg(args[n], XmNsliderSize, 1); ++n;
+    XtSetArg(args[n], XmNvalue, 0); ++n;
+    XtSetArg(args[n], XmNincrement, 1); ++n;
+    XtSetArg(args[n], XmNpageIncrement, 1); ++n;
+    hScrollBar = XmCreateScrollBar(scrolledWin, "hScrollBar", args, n);
+    XtManageChild(hScrollBar);
+    XtAddCallback(hScrollBar, XmNvalueChangedCallback,
+		    &hScrollChangeCbk, (XtPointer)this);
 #ifndef DISABLE_SMOOTH_SCROLL
-  XtAddCallback(hScrollBar, XmNdragCallback,
-		&hScrollDragCbk, (XtPointer)this);
+    XtAddCallback(hScrollBar, XmNdragCallback,
+		    &hScrollDragCbk, (XtPointer)this);
 #endif
-  n = 0;
-  XtSetArg(args[n], XmNorientation, XmVERTICAL); ++n;
-  XtSetArg(args[n], XmNminimum, 0); ++n;
-  XtSetArg(args[n], XmNmaximum, 1); ++n;
-  XtSetArg(args[n], XmNsliderSize, 1); ++n;
-  XtSetArg(args[n], XmNvalue, 0); ++n;
-  XtSetArg(args[n], XmNincrement, 1); ++n;
-  XtSetArg(args[n], XmNpageIncrement, 1); ++n;
-  vScrollBar = XmCreateScrollBar(scrolledWin, "vScrollBar", args, n);
-  XtManageChild(vScrollBar);
-  XtAddCallback(vScrollBar, XmNvalueChangedCallback,
-		&vScrollChangeCbk, (XtPointer)this);
+    n = 0;
+    XtSetArg(args[n], XmNorientation, XmVERTICAL); ++n;
+    XtSetArg(args[n], XmNminimum, 0); ++n;
+    XtSetArg(args[n], XmNmaximum, 1); ++n;
+    XtSetArg(args[n], XmNsliderSize, 1); ++n;
+    XtSetArg(args[n], XmNvalue, 0); ++n;
+    XtSetArg(args[n], XmNincrement, 1); ++n;
+    XtSetArg(args[n], XmNpageIncrement, 1); ++n;
+    vScrollBar = XmCreateScrollBar(scrolledWin, "vScrollBar", args, n);
+    XtManageChild(vScrollBar);
+    XtAddCallback(vScrollBar, XmNvalueChangedCallback,
+		    &vScrollChangeCbk, (XtPointer)this);
 #ifndef DISABLE_SMOOTH_SCROLL
-  XtAddCallback(vScrollBar, XmNdragCallback,
-		&vScrollDragCbk, (XtPointer)this);
+    XtAddCallback(vScrollBar, XmNdragCallback,
+		    &vScrollDragCbk, (XtPointer)this);
 #endif
-
+  }
   // create the drawing area
   n = 0;
   XtSetArg(args[n], XmNshadowType, XmSHADOW_IN); ++n;
@@ -1459,7 +1460,7 @@
   XtSetArg(args[n], XmNsliderSize, drawAreaWidth); ++n;
   XtSetArg(args[n], XmNincrement, 16); ++n;
   XtSetArg(args[n], XmNpageIncrement, drawAreaWidth); ++n;
-  XtSetValues(hScrollBar, args, n);
+  if (hScrollBar) XtSetValues(hScrollBar, args, n);
 
   if (pages->getLength() > 0) {
     if (continuousMode) {
@@ -1479,7 +1480,7 @@
   XtSetArg(args[n], XmNsliderSize, drawAreaHeight); ++n;
   XtSetArg(args[n], XmNincrement, 16); ++n;
   XtSetArg(args[n], XmNpageIncrement, drawAreaHeight); ++n;
-  XtSetValues(vScrollBar, args, n);
+  if (vScrollBar) XtSetValues(vScrollBar, args, n);
 }
 
 void XPDFCore::setCursor(Cursor cursor) {
